src/scrot.c

```diff
@@ -581,22 +581,24 @@ static Bool scrotXEventVisibility(Display *dpy, XEvent *ev, XPointer arg)
 static char *imPrintf(char *str, struct tm *tm, char *filenameIM,
     char *filenameThumb, Imlib_Image im)
 {
-    char *c;
     char buf[20];
     Stream ret = {0};
     long hostNameMax = 0;
     char strf[4096];
+    size_t strfLen;
     char *tmp;
     struct stat st;
 
-    if (strftime(strf, 4095, str, tm) == 0)
+    strfLen = strftime(strf, 4095, str, tm);
+    if (strfLen == 0)
         errx(EXIT_FAILURE, "strftime returned 0");
 
     imlib_context_set_image(im);
-    for (c = strf; *c != '\0'; c++) {
-        if (*c == '$') {
-            c++;
-            switch (*c) {
+    for (size_t i = 0; i < strfLen; ++i) {
+        char ch = strf[i];
+        if (ch == '$' && (i + 1) < strfLen) {
```

Comment:
Could also just use `*c == '$' && c[1] != '\0'` to minimize the diff.

I'm fine either ways, but using index seemed like a better idea.